pennylane/capture/measure.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the measure primitive for performing measurements in plxpr.
+"""
+from functools import lru_cache
+
+import pennylane as qml  # need shots class without circular dependency issues
+
+has_jax = True
+try:
+    import jax
+except ImportError:
+    has_jax = False
+
+
+@lru_cache
+def _get_measure_primitive():
+    if not has_jax:
+        raise ImportError("jax is required to create the measure primitive.")
+
+    if jax.config.jax_enable_x64:
+        dtype_map = {
+            float: jax.numpy.float64,
+            int: jax.numpy.int64,
+            complex: jax.numpy.complex128,
+        }
+    else:
+        dtype_map = {
+            float: jax.numpy.float32,
+            int: jax.numpy.int32,
+            complex: jax.numpy.complex64,
+        }
```

Comment:
Very nice to handle this in a single central place and allow measurements to be JAX-agnostic in the dtype they return :+1: